mlflow/store/tracking/file_store.py

```diff
@@ -588,6 +589,13 @@ def create_run(self, experiment_id, user_id, start_time, tags):
         mkdir(run_dir, FileStore.METRICS_FOLDER_NAME)
         mkdir(run_dir, FileStore.PARAMS_FOLDER_NAME)
         mkdir(run_dir, FileStore.ARTIFACTS_FOLDER_NAME)
+        if isinstance(tags, list):
+            if MLFLOW_RUN_NAME not in [tag.key for tag in tags]:
+                tags.append(RunTag(MLFLOW_RUN_NAME, _generate_random_name()))
+        # cli sends tags to the backend store as a dict
+        elif isinstance(tags, dict) and not tags:
+            if MLFLOW_RUN_NAME not in tags.keys():
+                tags = [RunTag(MLFLOW_RUN_NAME, _generate_random_name())]
```

Comment:
Are you sure? It's not clear to me how line 592 would work if that were the case, unless the dict keys are `RunTag` entities, in which case we should fix the CLI behavior rather than expanding the type handling for FileStore.

Reply:
100% due to trying to address a test failure (the test config in the unit test was incorrect). Fixed.